File: Warehousing.UI/src/app/admin/models/product.ts
import { Category } from "./category";
import { Inventory } from "./Inventory";
import { Store } from "./store";
import { SubCategory } from "./SubCategory";
import { Unit } from "./unit";
import { InventoryTransaction } from "./inventoryTransaction";
import { OrderItemDto } from "./OrderItemDto";
import { ProductVariant } from "./ProductVariant";

export interface Product {
    id: number;
    code: string;
    nameEn: string | null;
    nameAr: string;
    description: string | null;
    openingBalance: number | null;
    reorderLevel: number | null;
    imagePath: string | null;
    isActive: boolean;
    quantityInStock: number;
    costPrice: number;
    sellingPrice: number;
    lastStockUpdateDate: string;
    subCategory: SubCategory | null;
    subCategoryId: number | null;
    unitId: number | null;
    unit: Unit | null;
    storeId: number | null;
    store: Store | null;
    inventories: Inventory[];
    transactions: InventoryTransaction[];
    orderItems: OrderItemDto[];
    variants: ProductVariant[]; // Product variants
    modifierGroups?: any[]; // Product modifier groups
    totalQuantity?: number; // Total quantity across all stores
    variantStockData?: { [key: string]: any }; // Variant stock data from API
}
export interface ProductDto {
    id: number | null;
    code: string | null;
    nameEn: string | null;
    nameAr: string;
    description: string | null;
    openingBalance: number | null;
    reorderLevel: number | null;
    costPrice: number;
    sellingPrice: number;
    isActive: boolean;
    subCategory: SubCategory | null;
    subCategoryId: number | null;
    unitId: number | null;
    unit: Unit | null;
    storeId: number | null;
    store: Store | null;
    image: File | null;
    imagePath: string | null;
    inventories: Inventory[]
}

export interface ProductPagination {
    products: Product[],
    totals: number
}

File: Warehousing.Data/Entities/Product.cs
namespace Warehousing.Data.Entities
{
    public class Product : BaseClass
    {
        public int Id { get; set; }
        public string Code { get; set; } = string.Empty;
        public string? NameEn { get; set; } = string.Empty;
        public string NameAr { get; set; } = string.Empty;
        public string? Description { get; set; } = string.Empty;
        public string? ImagePath { get; set; } = string.Empty;
        public bool IsActive { get; set; } = true;
        public decimal CostPrice { get; set; }
        public decimal SellingPrice { get; set; }

        // FKs
        public SubCategory? SubCategory { get; set; }
        public int? SubCategoryId { get; set; }
        public int? UnitId { get; set; }
        public Unit? Unit { get; set; }

        // Navigation properties
        public ICollection<Inventory> Inventories { get; set; } = new List<Inventory>();
        public ICollection<InventoryTransaction> Transactions { get; set; } = new List<InventoryTransaction>();
        public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
        
        // Recipe support (future)
        public ICollection<ProductRecipe> RecipeAsParent { get; set; } = new List<ProductRecipe>();
        public ICollection<ProductRecipe> RecipeAsComponent { get; set; } = new List<ProductRecipe>();
        public ICollection<StoreTransferItem> TransferItems { get; set; } = new List<StoreTransferItem>();
    }
}